cmd/provisioner-localpv/app/helper_hostpath.go

```diff
@@ -106,59 +110,56 @@ func (pOpts *HelperPodOptions) validate() error {
 
 //validateLimits check that the limits to setup qouta are valid
 func (pOpts *HelperPodOptions) validateLimits() error {
-	if pOpts.bsoft == "0k" &&
-		pOpts.bhard == "0k" {
+	if pOpts.softLimitGrace == "0k" &&
+		pOpts.hardLimitGrace == "0k" {
 		return errors.Errorf("both limits cannot be 0")
 	}
 
-	if pOpts.bsoft == "0k" ||
-		pOpts.bhard == "0k" {
+	if pOpts.softLimitGrace == "0k" ||
+		pOpts.hardLimitGrace == "0k" {
 		return nil
 	}
 
-	if len(pOpts.bsoft) > len(pOpts.bhard) ||
-		(len(pOpts.bsoft) == len(pOpts.bhard) &&
-			pOpts.bsoft > pOpts.bhard) {
+	if len(pOpts.softLimitGrace) > len(pOpts.hardLimitGrace) ||
+		(len(pOpts.softLimitGrace) == len(pOpts.hardLimitGrace) &&
+			pOpts.softLimitGrace > pOpts.hardLimitGrace) {
 		return errors.Errorf("hard limit cannot be smaller than soft limit")
 	}
 
 	return nil
 }
 
 //converToK converts the limits to kilobytes
-func convertToK(limit string) (string, error) {
+func convertToK(limit string, pvcStorage int64) (string, error) {
 
 	if len(limit) == 0 {
 		return "0k", nil
```

Comment:
Instead of "OK", you can return null(which would signify empty value)

Reply:
its not "OK" it is "0k" , it is required because bsoft or bhard doesn't accept null

Reply:
So instead of `0K` return empty string i.e "". In that case we'll have better picture that the values were empty and not provided as 0.

Reply:
the command will give problem it only accepts size , if you do `bsoft=0k` that means soft limit will not be applied, there is no limit as `0k`

Reply:
I don't think that will reach that stage. Since if both the limits are empty then the validateLimits() will return error and we will not reach the stage of executing the quota pod. Here you are not checking the `0` value but checking the length of the limit string which if 0 means the user did not provide any value for that limit.

Reply:
Either you can address this or else I am fine with this also.

Reply:
If we address this code will unnecessarily become complex and for consistency , I think this is fine